configure.ac

```diff
@@ -1635,6 +1635,12 @@ SQUID_CHECK_LIB_WORKS(ldap,[
     AC_CHECK_HEADERS(mozldap/ldap.h)
     SQUID_CHECK_LDAP_API
   ])
+  # We can still rely on deprecated APIs on MacOS
+])
```

Comment:
The checks performed to detect a working library involve several compile tests, which could fail. Even if the header ones do not prevent `LIBLDAP_LIBS` being setup and used. The build would break later citing missing includes when they would build fine with the flag added.

I cannot attach this to the correct line. Please add the following `AS_IF` line between the two quoted.
```
    LIBS="$LIBLDAP_PATH $LIBPTHREADS $LIBS"
    AS_IF([test "x$squid_host_os" = "xdarwin"],[CXXFLAGS="-Wno-deprecated-declarations $CXXFLAGS"])
    PKG_CHECK_MODULES([LIBLDAP],[ldap],[],[
```

Reply:
This isn't portable, that flag is currently usupported by clang.
I think what we need to do is:
- when we detect compiler flags early in configure, detect (but not activate) this.
- then here rely on the detected value up there and use it if needed

Reply:
This is implemented in commit d46cc2b2dd

Reply:
FYI, github does not recognize that hash ID so I cannot check the menthioned change.

Reply:
> FYI, github does not recognize that hash ID so I cannot check the mentioned change.

Perhaps Francesco forgot to push his changes? IMO, it is usually best to post comments _after_ you push. That way, there is less confusion for the comment reader, and the comment author gets visual feedback if their commit SHAs are incorrect. One can write comments before pushing branch changes but post those comments after the push, but that requires even more discipline/care; IME, GitHub will not render links to SHAs unless an earlier written comment is updated after the push.
 
>  One need to copy the entire long-form hash or use github UI to locate the correct hash manually and copy that one instead.

IME, the situation is much simpler:  The short SHA needs to be known to GitHub (push first!) and long enough for GitHub to show a link to a single commit; seven digits is the current minimum AFAICT. I routinely post my 7-digit short-form SHAs in Squid comments, and they usually work fine. Longer short SHAs will have an even greater chance of success because they are more likely to be unique, of course.

Here is a test with this PR first commit SHA shortened to various lengths:

* six digits (this is the only test line that does not work today): 3b9e56
* seven digits: 3b9e563
* eight digits: 3b9e563a
* sixteen digits: 3b9e563ac88a80b4
* seventeen digits: 3b9e563ac88a80b48
* all forty digits: 3b9e563ac88a80b48e18d92a874ae79c55188328

As Squid repository gets more commits, using eight digits for short SHAs may become routinely necessary to avoid many conflicts. For example, they say that Linux kernel uses at least 11 digits, and git (repository) itself needs at least 10. See also: https://stackoverflow.com/questions/18134627/how-much-of-a-git-sha-is-generally-considered-necessary-to-uniquely-identify-a